contrib/babelfishpg_common/src/sqlvariant.c

```diff
@@ -79,6 +83,16 @@ sqlvariantin(PG_FUNCTION_ARGS)
 	Oid			typIOParam;
 	svhdr_5B_t *svhdr;
 
+	/*
+	 * Input as a bytea instead if it is logical replication applyworker.
+	 *    IsLogicalWorker() is sufficient for native PG applyworker but will
+	 *    not work with external providers like pglogical, so will rely on
+	 *    SessionReplicationRole being replica since most of the providers seem
+	 *    to set this GUC.
+	 */
+	if (IsLogicalWorker() || SessionReplicationRole == SESSION_REPLICATION_ROLE_REPLICA)
```

Comment:
it would be cleaner to add wrapper/macro functions to identify logical workers (source and target)

Reply:
Makes sense, added MACROS.